--- a/gfx/skia/moz.build
+++ b/gfx/skia/moz.build
@@ -558,7 +558,7 @@
     SOURCES += [
         'skia/src/ports/SkTypeface_mac_ct.cpp',
     ]
-if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'gtk':
+if CONFIG['MOZ_WIDGET_TOOLKIT'] in ('exos', 'gtk'):
     UNIFIED_SOURCES += [
         'skia/src/ports/SkLog_stdio.cpp',
         'skia/src/ports/SkOSFile_posix.cpp',
@@ -650,7 +650,7 @@
     "/gfx/harfbuzz/src",
 ]
 
-if CONFIG['MOZ_WIDGET_TOOLKIT'] in ('gtk', 'android'):
+if CONFIG['MOZ_WIDGET_TOOLKIT'] in ('android', 'exos', 'gtk'):
     LOCAL_INCLUDES += [
         "/gfx/cairo/cairo/src",
     ]
