--- a/gfx/2d/NativeFontResourceFreeType.cpp
+++ b/gfx/2d/NativeFontResourceFreeType.cpp
@@ -33,7 +33,7 @@
   return resource.forget();
 }
 
-#ifdef MOZ_WIDGET_ANDROID
+#if defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
 already_AddRefed<NativeFontResourceFreeType> NativeFontResourceFreeType::Create(
     const uint8_t* aFontData, uint32_t aDataLength, FT_Library aFTLibrary) {
   return CreateInternal<NativeFontResourceFreeType>(aFontData, aDataLength,
