--- a/gfx/2d/moz.build
+++ b/gfx/2d/moz.build
@@ -87,7 +87,7 @@
     ]
     DEFINES["WIN32"] = True
 
-if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "gtk"):
+if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "exos", "gtk"):
     EXPORTS.mozilla.gfx += [
         "UnscaledFontFreeType.h",
     ]
@@ -101,7 +101,7 @@
         "ScaledFontFontconfig.cpp",
     ]
 
-if CONFIG["MOZ_WIDGET_TOOLKIT"] == "android":
+if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "exos"):
     SOURCES += [
         "ScaledFontFreeType.cpp",
     ]
@@ -193,12 +193,12 @@
 for var in ("USE_CAIRO", "MOZ2D_HAS_MOZ_CAIRO"):
     DEFINES[var] = True
 
-if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "gtk"):
+if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "exos", "gtk"):
     DEFINES["MOZ_ENABLE_FREETYPE"] = True
 
 COMPILE_FLAGS["WARNINGS_CXXFLAGS"] += ["-Werror=switch"]
 
-if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "gtk"):
+if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("android", "exos", "gtk"):
     CXXFLAGS += CONFIG["CAIRO_FT_CFLAGS"]
 
 LOCAL_INCLUDES += CONFIG["SKIA_INCLUDES"]
