--- a/gfx/2d/Factory.cpp
+++ b/gfx/2d/Factory.cpp
@@ -27,7 +27,7 @@
 #  include "UnscaledFontFreeType.h"
 #endif
 
-#ifdef MOZ_WIDGET_ANDROID
+#if defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
 #  include "NativeFontResourceFreeType.h"
 #  include "ScaledFontFreeType.h"
 #  include "UnscaledFontFreeType.h"
@@ -412,7 +412,7 @@
     case FontType::FONTCONFIG:
       return NativeFontResourceFontconfig::Create(
           aData, aSize, static_cast<FT_Library>(aFontContext));
-#elif defined(MOZ_WIDGET_ANDROID)
+#elif defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
     case FontType::FREETYPE:
       return NativeFontResourceFreeType::Create(
           aData, aSize, static_cast<FT_Library>(aFontContext));
@@ -440,7 +440,7 @@
     case FontType::FONTCONFIG:
       return UnscaledFontFontconfig::CreateFromFontDescriptor(
           aData, aDataLength, aIndex);
-#elif defined(MOZ_WIDGET_ANDROID)
+#elif defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
     case FontType::FREETYPE:
       return UnscaledFontFreeType::CreateFromFontDescriptor(aData, aDataLength,
                                                             aIndex);
@@ -470,7 +470,7 @@
 }
 #endif
 
-#ifdef MOZ_WIDGET_ANDROID
+#if defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
 already_AddRefed<ScaledFont> Factory::CreateScaledFontForFreeTypeFont(
     const RefPtr<UnscaledFont>& aUnscaledFont, Float aSize,
     RefPtr<SharedFTFace> aFace, bool aApplySyntheticBold) {
