--- a/gfx/2d/NativeFontResourceFreeType.h
+++ b/gfx/2d/NativeFontResourceFreeType.h
@@ -18,7 +18,7 @@
  public:
   MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(NativeFontResourceFreeType, override)
 
-#ifdef MOZ_WIDGET_ANDROID
+#if defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
   static already_AddRefed<NativeFontResourceFreeType> Create(
       const uint8_t* aFontData, uint32_t aDataLength,
       FT_Library aFTLibrary = nullptr);
