--- a/gfx/2d/2D.h
+++ b/gfx/2d/2D.h
@@ -34,7 +34,7 @@
 #include "mozilla/gfx/FontData.h"
 #include "nsRegionFwd.h"
 
-#if defined(MOZ_WIDGET_ANDROID) || defined(MOZ_WIDGET_GTK)
+#if defined(MOZ_WIDGET_ANDROID) || defined(MOZ_WIDGET_GTK) || defined(XP_EXOS)
 #  ifndef MOZ_ENABLE_FREETYPE
 #    define MOZ_ENABLE_FREETYPE
 #  endif
@@ -2301,7 +2301,7 @@
       RefPtr<SharedFTFace> aFace, FcPattern* aPattern);
 #endif
 
-#ifdef MOZ_WIDGET_ANDROID
+#if defined(MOZ_WIDGET_ANDROID) || defined(XP_EXOS)
   static already_AddRefed<ScaledFont> CreateScaledFontForFreeTypeFont(
       const RefPtr<UnscaledFont>& aUnscaledFont, Float aSize,
       RefPtr<SharedFTFace> aFace, bool aApplySyntheticBold = false);
