--- a/js/src/wasm/WasmBuiltins.cpp
+++ b/js/src/wasm/WasmBuiltins.cpp
@@ -2165,7 +2165,7 @@
   }
 };
 
-MOZ_RUNINIT Mutex initBuiltinThunks(mutexid::WasmInitBuiltinThunks);
+MOZ_RUNINIT js::Mutex initBuiltinThunks(mutexid::WasmInitBuiltinThunks);
 mozilla::Atomic<const BuiltinThunks*> builtinThunks;
 
 bool wasm::EnsureBuiltinThunksInitialized() {
@@ -2175,7 +2175,7 @@
 
 bool wasm::EnsureBuiltinThunksInitialized(
     AutoMarkJitCodeWritableForThread& writable) {
-  LockGuard<Mutex> guard(initBuiltinThunks);
+  LockGuard<js::Mutex> guard(initBuiltinThunks);
   if (builtinThunks) {
     return true;
   }
