--- a/js/src/shell/js.cpp
+++ b/js/src/shell/js.cpp
@@ -1750,11 +1750,11 @@
 
 // Telemetry can be executed from multiple threads, and the callback is
 // responsible to avoid contention on the recorded telemetry data.
-class MOZ_RAII AutoLockTelemetry : public LockGuard<Mutex> {
-  using Base = LockGuard<Mutex>;
+class MOZ_RAII AutoLockTelemetry : public LockGuard<js::Mutex> {
+  using Base = LockGuard<js::Mutex>;
 
-  Mutex& getMutex() {
-    static Mutex mutex(mutexid::ShellTelemetry);
+  js::Mutex& getMutex() {
+    static js::Mutex mutex(mutexid::ShellTelemetry);
     return mutex;
   }
 
@@ -1905,9 +1905,9 @@
 }
 
 // Use Counter introspection
-MOZ_RUNINIT static Mutex useCounterLock(mutexid::ShellUseCounters);
-class MOZ_RAII AutoLockUseCounters : public LockGuard<Mutex> {
-  using Base = LockGuard<Mutex>;
+MOZ_RUNINIT static js::Mutex useCounterLock(mutexid::ShellUseCounters);
+class MOZ_RAII AutoLockUseCounters : public LockGuard<js::Mutex> {
+  using Base = LockGuard<js::Mutex>;
 
  public:
   AutoLockUseCounters() : Base(useCounterLock) {}
@@ -4752,12 +4752,12 @@
 }
 
 // Workers can spawn other workers, so we need a lock to access workerThreads.
-static Mutex* workerThreadsLock = nullptr;
+static js::Mutex* workerThreadsLock = nullptr;
 MOZ_RUNINIT static Vector<UniquePtr<js::Thread>, 0, SystemAllocPolicy>
     workerThreads;
 
-class MOZ_RAII AutoLockWorkerThreads : public LockGuard<Mutex> {
-  using Base = LockGuard<Mutex>;
+class MOZ_RAII AutoLockWorkerThreads : public LockGuard<js::Mutex> {
+  using Base = LockGuard<js::Mutex>;
 
  public:
   AutoLockWorkerThreads() : Base(*workerThreadsLock) {
@@ -4790,7 +4790,7 @@
   }
 
   if (!workerThreadsLock) {
-    workerThreadsLock = js_new<Mutex>(mutexid::ShellWorkerThreads);
+    workerThreadsLock = js_new<js::Mutex>(mutexid::ShellWorkerThreads);
     if (!workerThreadsLock) {
       ReportOutOfMemory(cx);
       return false;
@@ -4870,7 +4870,7 @@
     }
   }
   {
-    LockGuard<Mutex> guard(sc->watchdogLock);
+    LockGuard<js::Mutex> guard(sc->watchdogLock);
     TimeStamp toWakeup = TimeStamp::Now() + duration;
     for (;;) {
       sc->sleepWakeup.wait_for(guard, duration);
@@ -4899,7 +4899,7 @@
   Maybe<Thread> thread;
 
   {
-    LockGuard<Mutex> guard(sc->watchdogLock);
+    LockGuard<js::Mutex> guard(sc->watchdogLock);
     std::swap(sc->watchdogThread, thread);
     if (thread) {
       // The watchdog thread becoming Nothing is its signal to exit.
@@ -4919,7 +4919,7 @@
   ShellContext* sc = GetShellContext(cx);
 
   {
-    LockGuard<Mutex> guard(sc->watchdogLock);
+    LockGuard<js::Mutex> guard(sc->watchdogLock);
     while (sc->watchdogThread) {
       auto now = TimeStamp::Now();
       if (sc->watchdogTimeout && now >= sc->watchdogTimeout.value()) {
@@ -4957,7 +4957,7 @@
   ShellContext* sc = GetShellContext(cx);
 
   if (t <= 0) {
-    LockGuard<Mutex> guard(sc->watchdogLock);
+    LockGuard<js::Mutex> guard(sc->watchdogLock);
     sc->watchdogTimeout = Nothing();
     return true;
   }
@@ -4968,7 +4968,7 @@
 
   auto interval = TimeDuration::FromSeconds(t);
   auto timeout = TimeStamp::Now() + interval;
-  LockGuard<Mutex> guard(sc->watchdogLock);
+  LockGuard<js::Mutex> guard(sc->watchdogLock);
   if (!sc->watchdogThread) {
     MOZ_ASSERT(!sc->watchdogTimeout);
     sc->watchdogThread.emplace();
