--- a/js/src/debugger/ExecutionTracer.cpp
+++ b/js/src/debugger/ExecutionTracer.cpp
@@ -26,7 +26,7 @@
 using namespace js;
 
 MOZ_RUNINIT mozilla::Vector<ExecutionTracer*> ExecutionTracer::globalInstances;
-MOZ_RUNINIT Mutex
+MOZ_RUNINIT js::Mutex
     ExecutionTracer::globalInstanceLock(mutexid::ExecutionTracerGlobalLock);
 
 // This is a magic value we write as the last 64 bits of a FunctionEnter event
@@ -200,7 +200,7 @@
 }
 
 void ExecutionTracer::onEnterFrame(JSContext* cx, AbstractFramePtr frame) {
-  LockGuard<Mutex> guard(bufferLock_);
+  LockGuard<js::Mutex> guard(bufferLock_);
 
   DebuggerFrameType type = GetFrameType(frame);
   if (type == DebuggerFrameType::Call) {
@@ -229,7 +229,7 @@
 }
 
 void ExecutionTracer::onLeaveFrame(JSContext* cx, AbstractFramePtr frame) {
-  LockGuard<Mutex> guard(bufferLock_);
+  LockGuard<js::Mutex> guard(bufferLock_);
 
   DebuggerFrameType type = GetFrameType(frame);
   if (type == DebuggerFrameType::Call) {
@@ -247,7 +247,7 @@
 
 template <typename CharType, JS::TracerStringEncoding Encoding>
 void ExecutionTracer::onEnterLabel(const CharType* eventType) {
-  LockGuard<Mutex> guard(bufferLock_);
+  LockGuard<js::Mutex> guard(bufferLock_);
 
   inlineData_.beginWritingEntry();
   inlineData_.write(uint8_t(InlineEntryType::LabelEnter));
@@ -258,7 +258,7 @@
 
 template <typename CharType, JS::TracerStringEncoding Encoding>
 void ExecutionTracer::onLeaveLabel(const CharType* eventType) {
-  LockGuard<Mutex> guard(bufferLock_);
+  LockGuard<js::Mutex> guard(bufferLock_);
 
   inlineData_.beginWritingEntry();
   inlineData_.write(uint8_t(InlineEntryType::LabelLeave));
@@ -545,7 +545,7 @@
 bool ExecutionTracer::getNativeTrace(
     JS::ExecutionTrace::TracedJSContext& context,
     TracingScratchBuffer& scratchBuffer, mozilla::Vector<char>& stringBuffer) {
-  LockGuard<Mutex> guard(bufferLock_);
+  LockGuard<js::Mutex> guard(bufferLock_);
 
   if (!readOutOfLineEntries(context.scriptUrls, context.atoms,
                             context.shapeSummaries, scratchBuffer,
@@ -565,7 +565,7 @@
 }
 
 bool ExecutionTracer::getNativeTraceForAllContexts(JS::ExecutionTrace& trace) {
-  LockGuard<Mutex> guard(globalInstanceLock);
+  LockGuard<js::Mutex> guard(globalInstanceLock);
   TracingScratchBuffer scratchBuffer;
   for (ExecutionTracer* tracer : globalInstances) {
     JS::ExecutionTrace::TracedJSContext* context = nullptr;
