--- a/third_party/rust/rustix/src/fs/fd.rs
+++ b/third_party/rust/rustix/src/fs/fd.rs
@@ -10,7 +10,7 @@
 #[cfg(not(any(
     netbsdlike,
     target_os = "dragonfly",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "nto",
     target_os = "redox",
@@ -18,7 +18,7 @@
 )))]
 use backend::fs::types::FallocateFlags;
 #[cfg(not(any(
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "solaris",
     target_os = "vita",
@@ -30,7 +30,7 @@
 use backend::fs::types::Stat;
 #[cfg(not(any(
     solarish,
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "haiku",
     target_os = "horizon",
     target_os = "netbsd",
@@ -168,7 +168,7 @@
 /// [Linux]: https://man7.org/linux/man-pages/man2/fstatfs.2.html
 #[cfg(not(any(
     solarish,
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "haiku",
     target_os = "horizon",
     target_os = "netbsd",
@@ -210,7 +210,7 @@
 ///
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/futimens.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/utimensat.2.html
-#[cfg(not(any(target_os = "espidf", target_os = "horizon", target_os = "vita")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "horizon", target_os = "vita")))]
 #[inline]
 pub fn futimens<Fd: AsFd>(fd: Fd, times: &Timestamps) -> io::Result<()> {
     backend::fs::syscalls::futimens(fd.as_fd(), times)
@@ -234,7 +234,7 @@
 #[cfg(not(any(
     netbsdlike,
     target_os = "dragonfly",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "nto",
     target_os = "redox",
@@ -276,7 +276,7 @@
 #[cfg(not(any(
     apple,
     target_os = "dragonfly",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "haiku",
     target_os = "horizon",
     target_os = "redox",
@@ -307,7 +307,7 @@
 ///
 /// [Linux]: https://man7.org/linux/man-pages/man2/flock.2.html
 #[cfg(not(any(
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "solaris",
     target_os = "vita",
