--- a/third_party/rust/rustix/src/backend/libc/event/types.rs
+++ b/third_party/rust/rustix/src/backend/libc/event/types.rs
@@ -4,7 +4,7 @@
     linux_kernel,
     target_os = "freebsd",
     target_os = "illumos",
-    target_os = "espidf"
+    any(target_os = "espidf", target_os = "exos")
 ))]
 use bitflags::bitflags;
 
@@ -12,7 +12,7 @@
     linux_kernel,
     target_os = "freebsd",
     target_os = "illumos",
-    target_os = "espidf"
+    any(target_os = "espidf", target_os = "exos")
 ))]
 bitflags! {
     /// `EFD_*` flags for use with [`eventfd`].
@@ -22,13 +22,13 @@
     #[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
     pub struct EventfdFlags: u32 {
         /// `EFD_CLOEXEC`
-        #[cfg(not(target_os = "espidf"))]
+        #[cfg(not(any(target_os = "espidf", target_os = "exos")))]
         const CLOEXEC = bitcast!(c::EFD_CLOEXEC);
         /// `EFD_NONBLOCK`
-        #[cfg(not(target_os = "espidf"))]
+        #[cfg(not(any(target_os = "espidf", target_os = "exos")))]
         const NONBLOCK = bitcast!(c::EFD_NONBLOCK);
         /// `EFD_SEMAPHORE`
-        #[cfg(not(target_os = "espidf"))]
+        #[cfg(not(any(target_os = "espidf", target_os = "exos")))]
         const SEMAPHORE = bitcast!(c::EFD_SEMAPHORE);
 
         /// <https://docs.rs/bitflags/*/bitflags/#externally-defined-flags>
