--- a/third_party/angle/preprocessor_gn/moz.build
+++ b/third_party/angle/preprocessor_gn/moz.build
@@ -109,7 +109,7 @@
         "dl"
     ]
 
-if CONFIG["OS_TARGET"] == "OpenBSD":
+if CONFIG["OS_TARGET"] in ("OpenBSD", "EXOS"):
 
     DEFINES["EGL_NO_X11"] = True
     DEFINES["USE_GLIB"] = "1"
@@ -180,7 +180,7 @@
 
     DEFINES["_DEBUG"] = True
 
-if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "OpenBSD":
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] in ("OpenBSD", "EXOS"):
 
     DEFINES["_DEBUG"] = True
 
