--- a/third_party/rust/rustix/src/backend/libc/termios/syscalls.rs
+++ b/third_party/rust/rustix/src/backend/libc/termios/syscalls.rs
@@ -13,19 +13,19 @@
 #[cfg(not(any(target_os = "fuchsia", target_os = "wasi")))]
 use crate::ffi::CStr;
 #[cfg(any(
-    not(target_os = "espidf"),
+    not(any(target_os = "espidf", target_os = "exos")),
     not(any(target_os = "fuchsia", target_os = "wasi"))
 ))]
 use core::mem::MaybeUninit;
 #[cfg(not(target_os = "wasi"))]
 use {crate::io, crate::pid::Pid};
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 use {
     crate::termios::{Action, OptionalActions, QueueSelector, Termios, Winsize},
     crate::utils::as_mut_ptr,
 };
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 pub(crate) fn tcgetattr(fd: BorrowedFd<'_>) -> io::Result<Termios> {
     // On Linux, use `TCGETS2`, and fall back to `TCGETS` if needed.
     #[cfg(linux_kernel)]
@@ -184,7 +184,7 @@
     unsafe { ret(c::tcsetpgrp(borrowed_fd(fd), pid.as_raw_nonzero().get())) }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 pub(crate) fn tcsetattr(
     fd: BorrowedFd<'_>,
     optional_actions: OptionalActions,
@@ -327,17 +327,17 @@
     unsafe { ret(c::tcsendbreak(borrowed_fd(fd), 0)) }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 pub(crate) fn tcdrain(fd: BorrowedFd<'_>) -> io::Result<()> {
     unsafe { ret(c::tcdrain(borrowed_fd(fd))) }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 pub(crate) fn tcflush(fd: BorrowedFd<'_>, queue_selector: QueueSelector) -> io::Result<()> {
     unsafe { ret(c::tcflush(borrowed_fd(fd), queue_selector as _)) }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 pub(crate) fn tcflow(fd: BorrowedFd<'_>, action: Action) -> io::Result<()> {
     unsafe { ret(c::tcflow(borrowed_fd(fd), action as _)) }
 }
@@ -350,12 +350,12 @@
     }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "horizon", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "horizon", target_os = "wasi")))]
 pub(crate) fn tcsetwinsize(fd: BorrowedFd<'_>, winsize: Winsize) -> io::Result<()> {
     unsafe { ret(c::ioctl(borrowed_fd(fd), c::TIOCSWINSZ, &winsize)) }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "horizon", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "horizon", target_os = "wasi")))]
 pub(crate) fn tcgetwinsize(fd: BorrowedFd<'_>) -> io::Result<Winsize> {
     unsafe {
         let mut buf = MaybeUninit::<Winsize>::uninit();
@@ -368,7 +368,7 @@
     }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "nto", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "nto", target_os = "wasi")))]
 #[inline]
 pub(crate) fn set_speed(termios: &mut Termios, arbitrary_speed: u32) -> io::Result<()> {
     #[cfg(bsd)]
@@ -410,7 +410,7 @@
     }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 #[inline]
 pub(crate) fn set_output_speed(termios: &mut Termios, arbitrary_speed: u32) -> io::Result<()> {
     #[cfg(bsd)]
@@ -450,7 +450,7 @@
     }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 #[inline]
 pub(crate) fn set_input_speed(termios: &mut Termios, arbitrary_speed: u32) -> io::Result<()> {
     #[cfg(bsd)]
@@ -490,7 +490,7 @@
     }
 }
 
-#[cfg(not(any(target_os = "espidf", target_os = "nto", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "nto", target_os = "wasi")))]
 #[inline]
 pub(crate) fn cfmakeraw(termios: &mut Termios) {
     unsafe {
