--- a/third_party/angle/angle_translator_headers_gn/moz.build
+++ b/third_party/angle/angle_translator_headers_gn/moz.build
@@ -73,7 +73,7 @@
     DEFINES["__STDC_CONSTANT_MACROS"] = True
     DEFINES["__STDC_FORMAT_MACROS"] = True
 
-if CONFIG["OS_TARGET"] == "OpenBSD":
+if CONFIG["OS_TARGET"] in ("OpenBSD", "EXOS"):
 
     DEFINES["EGL_NO_X11"] = True
     DEFINES["USE_GLIB"] = "1"
@@ -140,7 +140,7 @@
 
     DEFINES["_DEBUG"] = True
 
-if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "OpenBSD":
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] in ("OpenBSD", "EXOS"):
 
     DEFINES["_DEBUG"] = True
 
