--- a/third_party/rust/rustix/src/io/fcntl.rs
+++ b/third_party/rust/rustix/src/io/fcntl.rs
@@ -99,7 +99,7 @@
 /// [illumos]: https://illumos.org/man/2/fcntl
 /// [glibc]: https://sourceware.org/glibc/manual/latest/html_node/Control-Operations.html#index-fcntl-function
 /// [file description]: https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap03.html#tag_03_258
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 #[inline]
 #[doc(alias = "F_DUPFD_CLOEXEC")]
 pub fn fcntl_dupfd_cloexec<Fd: AsFd>(fd: Fd, min: RawFd) -> io::Result<OwnedFd> {
@@ -135,7 +135,7 @@
 /// [illumos]: https://illumos.org/man/2/fcntl
 /// [glibc]: https://sourceware.org/glibc/manual/latest/html_node/Control-Operations.html#index-fcntl-function
 /// [file description]: https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap03.html#tag_03_258
-#[cfg(target_os = "espidf")]
+#[cfg(any(target_os = "espidf", target_os = "exos"))]
 #[inline]
 #[doc(alias = "F_DUPFD")]
 pub fn fcntl_dupfd<Fd: AsFd>(fd: Fd, min: RawFd) -> io::Result<OwnedFd> {
