--- a/third_party/rust/rustix/src/fs/fcntl.rs
+++ b/third_party/rust/rustix/src/fs/fcntl.rs
@@ -5,7 +5,7 @@
 
 #[cfg(not(any(
     target_os = "emscripten",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "fuchsia",
     target_os = "horizon",
     target_os = "redox",
@@ -92,7 +92,7 @@
 /// [Linux]: https://man7.org/linux/man-pages/man2/fcntl.2.html
 #[cfg(not(any(
     target_os = "emscripten",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "fuchsia",
     target_os = "horizon",
     target_os = "redox",
