--- a/third_party/rust/rustix/src/fs/at.rs
+++ b/third_party/rust/rustix/src/fs/at.rs
@@ -10,19 +10,19 @@
 
 use crate::buffer::Buffer;
 use crate::fd::OwnedFd;
-#[cfg(not(any(target_os = "espidf", target_os = "horizon", target_os = "vita")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "horizon", target_os = "vita")))]
 use crate::fs::Access;
-#[cfg(not(any(target_os = "espidf", target_os = "redox")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "redox")))]
 use crate::fs::AtFlags;
 #[cfg(apple)]
 use crate::fs::CloneFlags;
 #[cfg(any(linux_kernel, apple, target_os = "redox"))]
 use crate::fs::RenameFlags;
-#[cfg(not(target_os = "espidf"))]
+#[cfg(not(any(target_os = "espidf", target_os = "exos")))]
 use crate::fs::Stat;
-#[cfg(not(any(apple, target_os = "espidf", target_os = "vita", target_os = "wasi")))]
+#[cfg(not(any(apple, any(target_os = "espidf", target_os = "exos"), target_os = "vita", target_os = "wasi")))]
 use crate::fs::{Dev, FileType};
-#[cfg(not(any(target_os = "espidf", target_os = "wasi")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi")))]
 use crate::fs::{Gid, Uid};
 use crate::fs::{Mode, OFlags};
 use crate::{backend, io, path};
@@ -34,14 +34,14 @@
     alloc::vec::Vec,
     backend::fd::BorrowedFd,
 };
-#[cfg(not(any(target_os = "espidf", target_os = "vita")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "vita")))]
 use {crate::fs::Timestamps, crate::timespec::Nsecs};
 
 /// `UTIME_NOW` for use with [`utimensat`].
 ///
 /// [`utimensat`]: crate::fs::utimensat
 #[cfg(not(any(
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "redox",
     target_os = "vita"
@@ -52,7 +52,7 @@
 ///
 /// [`utimensat`]: crate::fs::utimensat
 #[cfg(not(any(
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "redox",
     target_os = "vita"
@@ -209,7 +209,7 @@
 ///
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/linkat.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/linkat.2.html
-#[cfg(not(any(target_os = "espidf", target_os = "redox")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "redox")))]
 #[inline]
 pub fn linkat<P: path::Arg, Q: path::Arg, PFd: AsFd, QFd: AsFd>(
     old_dirfd: PFd,
@@ -243,7 +243,7 @@
 /// [`REMOVEDIR`]: AtFlags::REMOVEDIR
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/unlinkat.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/unlinkat.2.html
-#[cfg(not(any(target_os = "espidf", target_os = "redox")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "redox")))]
 #[inline]
 pub fn unlinkat<P: path::Arg, Fd: AsFd>(dirfd: Fd, path: P, flags: AtFlags) -> io::Result<()> {
     path.into_with_c_str(|path| backend::fs::syscalls::unlinkat(dirfd.as_fd(), path, flags))
@@ -348,7 +348,7 @@
 /// [Linux]: https://man7.org/linux/man-pages/man2/fstatat.2.html
 /// [`Mode::from_raw_mode`]: crate::fs::Mode::from_raw_mode
 /// [`FileType::from_raw_mode`]: crate::fs::FileType::from_raw_mode
-#[cfg(not(any(target_os = "espidf", target_os = "redox")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "redox")))]
 #[inline]
 #[doc(alias = "fstatat")]
 pub fn statat<P: path::Arg, Fd: AsFd>(dirfd: Fd, path: P, flags: AtFlags) -> io::Result<Stat> {
@@ -372,7 +372,7 @@
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/faccessat.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/faccessat.2.html
 #[cfg(not(any(
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "vita",
     target_os = "redox"
@@ -397,7 +397,7 @@
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/utimensat.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/utimensat.2.html
 #[cfg(not(any(
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "vita",
     target_os = "redox"
@@ -424,7 +424,7 @@
 ///
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/fchmodat.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/fchmodat.2.html
-#[cfg(not(any(target_os = "espidf", target_os = "wasi", target_os = "redox")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi", target_os = "redox")))]
 #[inline]
 #[doc(alias = "fchmodat")]
 pub fn chmodat<P: path::Arg, Fd: AsFd>(
@@ -465,7 +465,7 @@
 /// [Linux]: https://man7.org/linux/man-pages/man2/mknodat.2.html
 #[cfg(not(any(
     apple,
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "vita",
     target_os = "wasi",
@@ -492,7 +492,7 @@
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/mkfifoat.html
 #[cfg(not(any(
     apple,
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "horizon",
     target_os = "vita",
     target_os = "wasi",
@@ -512,7 +512,7 @@
 ///
 /// [POSIX]: https://pubs.opengroup.org/onlinepubs/9799919799/functions/fchownat.html
 /// [Linux]: https://man7.org/linux/man-pages/man2/fchownat.2.html
-#[cfg(not(any(target_os = "espidf", target_os = "wasi", target_os = "redox")))]
+#[cfg(not(any(any(target_os = "espidf", target_os = "exos"), target_os = "wasi", target_os = "redox")))]
 #[inline]
 #[doc(alias = "fchownat")]
 pub fn chownat<P: path::Arg, Fd: AsFd>(
