--- a/third_party/rust/rustix/src/backend/libc/thread/syscalls.rs
+++ b/third_party/rust/rustix/src/backend/libc/thread/syscalls.rs
@@ -11,7 +11,7 @@
     apple,
     freebsdlike,
     target_os = "emscripten",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "haiku",
     target_os = "openbsd",
     target_os = "redox",
@@ -49,7 +49,7 @@
     apple,
     target_os = "dragonfly",
     target_os = "emscripten",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "freebsd", // FreeBSD 12 has clock_nanosleep, but libc targets FreeBSD 11.
     target_os = "haiku",
     target_os = "horizon",
@@ -160,7 +160,7 @@
     apple,
     target_os = "dragonfly",
     target_os = "emscripten",
-    target_os = "espidf",
+    any(target_os = "espidf", target_os = "exos"),
     target_os = "freebsd", // FreeBSD 12 has clock_nanosleep, but libc targets FreeBSD 11.
     target_os = "haiku",
     target_os = "horizon",
