--- a/widget/exos/MediaKeysEventSourceFactory.cpp
+++ b/widget/exos/MediaKeysEventSourceFactory.cpp
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "MediaKeysEventSourceFactory.h"
+
+namespace mozilla {
+namespace widget {
+
+// EX-OS (Exilla) has no media keys to listen to.
+mozilla::dom::MediaControlKeySource* CreateMediaControlKeySource() {
+  return nullptr;
+}
+
+}  // namespace widget
+}  // namespace mozilla
