--- a/widget/exos/nsUserIdleServiceExos.h
+++ b/widget/exos/nsUserIdleServiceExos.h
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef nsUserIdleServiceExos_h_
+#define nsUserIdleServiceExos_h_
+
+#include "mozilla/AppShutdown.h"
+#include "nsUserIdleService.h"
+
+class nsUserIdleServiceExos : public nsUserIdleService {
+ public:
+  NS_INLINE_DECL_REFCOUNTING_INHERITED(nsUserIdleServiceExos,
+                                       nsUserIdleService)
+
+  bool PollIdleTime(uint32_t* aIdleTime) override;
+
+  static already_AddRefed<nsUserIdleServiceExos> GetInstance() {
+    RefPtr<nsUserIdleService> idleService = nsUserIdleService::GetInstance();
+    if (!idleService) {
+      // Avoid late instantiation or resurrection during shutdown.
+      if (mozilla::AppShutdown::IsInOrBeyond(
+              mozilla::ShutdownPhase::AppShutdownConfirmed)) {
+        return nullptr;
+      }
+      idleService = new nsUserIdleServiceExos();
+    }
+
+    return idleService.forget().downcast<nsUserIdleServiceExos>();
+  }
+
+ protected:
+  nsUserIdleServiceExos() {}
+  virtual ~nsUserIdleServiceExos() {}
+};
+
+#endif  // nsUserIdleServiceExos_h_
