--- a/widget/moz.build
+++ b/widget/moz.build
@@ -53,7 +53,7 @@
     "components.conf",
 ]
 
-if toolkit in ("android", "cocoa", "gtk", "uikit", "windows"):
+if toolkit in ("android", "cocoa", "exos", "gtk", "uikit", "windows"):
     DIRS += [toolkit]
 
 if toolkit == "cocoa":
@@ -328,7 +328,7 @@
 if CONFIG["MOZ_X11"] or CONFIG["MOZ_WAYLAND"]:
     DIRS += ["x11"]
 
-if toolkit in {"gtk", "cocoa", "windows", "android", "uikit"}:
+if toolkit in {"gtk", "cocoa", "windows", "android", "uikit", "exos"}:
     UNIFIED_SOURCES += [
         "nsBaseFilePicker.cpp",
     ]
@@ -338,7 +338,7 @@
         "nsBaseColorPicker.cpp",
     ]
 
-if toolkit in ("gtk", "windows", "cocoa", "uikit", "android"):
+if toolkit in ("gtk", "windows", "cocoa", "uikit", "android", "exos"):
     UNIFIED_SOURCES += [
         "nsNativeTheme.cpp",
     ]
